sdk/tests/wallet/output_preparation.rs

```diff
@@ -557,3 +559,192 @@ async fn prepare_nft_output_features_update() -> Result<()> {
+    let account = &accounts[0];
+    let addresses = &accounts[1].addresses().await?;
+    let address = addresses[0].address();
+
+    let rent_structure = account.client().get_rent_structure().await?;
+    let token_supply = account.client().get_token_supply().await?;
+
+    let balance = account.sync(None).await?;
+    let minimum_required_storage_deposit =
+        MinimumStorageDepositBasicOutput::new(rent_structure, token_supply).finish()?;
+
+    // Send away most balance so we can test with leaving dust
+    let output = account
+        .prepare_output(
+            OutputParams {
+                recipient_address: *address,
+                amount: balance.base_coin().available() - 63900,
+                assets: None,
+                features: None,
+                unlocks: None,
+                storage_deposit: None,
+            },
+            None,
+        )
+        .await?;
+    let transaction = account.send_outputs(vec![output], None).await?;
+    account
+        .retry_transaction_until_included(&transaction.transaction_id, None, None)
+        .await?;
+    let balance = account.sync(None).await?;
+
+    // 63900 left
+    let output = account
+        .prepare_output(
+            OutputParams {
+                recipient_address: *address,
+                amount: minimum_required_storage_deposit - 1, // Leave less than min. deposit
+                assets: None,
+                features: None,
+                unlocks: None,
+                storage_deposit: Some(StorageDeposit {
+                    return_strategy: Some(ReturnStrategy::Gift),
+                    use_excess_if_low: Some(true),
+                }),
+            },
+            None,
+        )
+        .await?;
+
+    // Check if the output has enough amount to cover the storage deposit
+    output.verify_storage_deposit(rent_structure, token_supply)?;
+    // The left over 21299 is too small to keep, so we donate it
+    assert_eq!(output.amount(), balance.base_coin().available());
+    // storage deposit gifted, only address unlock condition
+    assert_eq!(output.unlock_conditions().unwrap().len(), 1);
+
+    let result = account
+        .prepare_output(
+            OutputParams {
+                recipient_address: *address,
+                amount: minimum_required_storage_deposit - 1, // Leave less than min. deposit
+                assets: None,
+                features: None,
+                unlocks: None,
+                storage_deposit: Some(StorageDeposit {
+                    return_strategy: Some(ReturnStrategy::Return),
+                    use_excess_if_low: Some(true),
+                }),
+            },
+            None,
+        )
+        .await;
+    matches!(result, Err(iota_sdk::wallet::Error::InsufficientFunds{available, required}) if available == balance.base_coin().available() && required == balance.base_coin().available()+minimum_required_storage_deposit-1);
```

Comment:
This isn't doing anything, needs an assert

Reply:
Thanks, already wondered why it passes, but then forgot to update :carlos: